File: fontend/vite-project/src/services/speechService.js
import apiClient from "./apiClient";

// ─── Speech Service ───────────────────────────────────────────
// Gọi Azure Cognitive Speech để tổng hợp giọng nói (TTS)
// Dùng tại: V4 – Trang thuyết minh gian hàng (/booth/:boothId)

const speechService = {
  /**
   * Tạo audio TTS từ nội dung văn bản
   * Backend chọn voice phù hợp theo languageCode:
   *   vi → vi-VN-HoaiMyNeural
   *   en → en-US-JennyNeural
   *   ja → ja-JP-NanamiNeural
   *   ko → ko-KR-SunHiNeural
   *   zh → zh-CN-XiaoxiaoNeural
   *
   * @param {string} text          - Nội dung cần đọc
   * @param {string} languageCode  - "vi" | "en" | "ja" | "ko" | "zh"
   * Returns: { audioUrl: string } - URL file MP3 để frontend phát
   */
  generateAudio: (text, languageCode) =>
    apiClient.post("/speech/generate", { text, languageCode }),
};

export default speechService;

File: backend/Services/SpeechService.cs
namespace backend.Services;

public class SpeechService
{
    // TODO: Tích hợp Azure Cognitive Speech
    private static readonly Dictionary<string, string> VoiceMap = new()
    {
        { "vi", "vi-VN-HoaiMyNeural"    },
        { "en", "en-US-JennyNeural"     },
        { "ja", "ja-JP-NanamiNeural"    },
        { "ko", "ko-KR-SunHiNeural"     },
        { "zh", "zh-CN-XiaoxiaoNeural"  },
    };

    public Task<object> GenerateAudioAsync(string text, string languageCode)
    {
        // Mock: trả về URL giả — thay bằng Azure SDK thật
        var voice    = VoiceMap.GetValueOrDefault(languageCode, "vi-VN-HoaiMyNeural");
        var audioUrl = $"/audio/{languageCode}/{Guid.NewGuid()}.mp3";

        return Task.FromResult<object>(new { audioUrl, voice, languageCode });
    }
}

public class SpeechRequest
{
    public string Text         { get; set; } = string.Empty;
    public string LanguageCode { get; set; } = "vi";
}